src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractTypeImportCompletionProvider.cs

```diff
@@ -42,23 +44,34 @@ protected override async Task AddCompletionItemsAsync(CompletionContext completi
                 }
                 else
                 {
+                    // In case the caret is at the beginning of the file, take the root node.
+                    var aliasSymbols = GetAliasSymbolsName(
+                        syntaxContext.LeftToken.Parent ?? await syntaxContext.SyntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false),
+                        syntaxContext.SemanticModel,
+                        cancellationToken);
                     foreach (var items in itemsFromAllAssemblies)
                     {
-                        AddItems(items, completionContext, namespacesInScope, telemetryCounter);
+                        AddItems(items, completionContext, namespacesInScope, aliasSymbols, telemetryCounter);
                     }
                 }
 
                 telemetryCounter.Report();
             }
         }
 
-        private static void AddItems(ImmutableArray<CompletionItem> items, CompletionContext completionContext, HashSet<string> namespacesInScope, TelemetryCounter counter)
+        private static void AddItems(
+            ImmutableArray<CompletionItem> items,
+            CompletionContext completionContext,
+            HashSet<string> namespacesInScope,
+            HashSet<string> aliasSymbolsName,
+            TelemetryCounter counter)
         {
             counter.ReferenceCount++;
             foreach (var item in items)
             {
                 var containingNamespace = ImportCompletionItem.GetContainingNamespace(item);
-                if (!namespacesInScope.Contains(containingNamespace))
+                var fullyQualifiedName = ImportCompletionItem.GetFullyQualifiedNameForTypeItem(item);
```

Comment:
It'd be rare to have using alias in a document, we should check if the dictionary is empty before getting the FQN to avoid unnecessary allocation. Additionally, instead of a hashset of FQN, we may consider a multi-dictionary of <NamespaceName, TypeNames> for all aliases, so we can avoid string allocations altogehter even with using alias presents

Reply:
Done, use MultiDictionary to store namespace -> Type Name